### ISSUE TITLE
Determine if we can remove the `StructToStruct` calls in unstructured_util.go

### ISSUE THREAD

**Issue by juliev0** [numaproj/numaplane#456]
**Describe the bug**
By calling `StructToStruct` from `GetResource()` and `GetLiveResource()` functions, the `Generation` field is changing from an integer field to a float64 field. Would like to remove `StructToStruct` if it doesn't cause harm. Need to figure out if we can instead update tests.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We often sort issues this way to know what to prioritize.

**Comment by chandankumar4** [comment]
Observation: Before and after conversion of `unstruc.Object` via `util.StructToStruct()` method, here `rpu` type is changed from `int64` to `float64`

Before:
<img width="522" alt="Screenshot 2024-12-15 at 5 27 49 PM" src="https://github.com/user-attachments/assets/3779b4c1-a667-4b66-84ff-200746d009db" />

After:
<img width="522" alt="Screenshot 2024-12-15 at 5 28 05 PM" src="https://github.com/user-attachments/assets/3221ff6d-616e-4890-bdc9-209bfec2178d" />

**Comment by juliev0** [comment]
Are all numbers float64 after? Maybe json unmarshal just does that?

**Comment by chandankumar4** [comment]
> Are all numbers float64 after? Maybe json unmarshal just does that?

Yes, Json unmarshal does that, similar conversation going [here](https://forum.golangbridge.org/t/type-problem-in-json-conversion/19420), which is closed saying `JSON does only know floats`

**Comment by chandankumar4** [comment]
[This](https://github.com/numaproj/numaplane/blob/main/internal/controller/pipelinerollout/pipelinerollout_controller.go#L783) is where problem start, here it converts the `pipelineRollout.Spec.Pipeline.Spec` of type `runtime.RawExtension` to `pipelineSpec` of type `map[string]interface{}` and treat numeric value as float64, so everywhere we have to convert `int64` value to `float64` otherwise `reflect.DeepEqual()` fail the comparison due to type difference.
So there is two possible fix:
1. Fix the Json parser to convert number to int64 instead of float64
2. Fix `reflect.DeepEqual()` method to only compare value not type in case of number

**Comment by juliev0** [comment]
Which call to `reflect.DeepEqual()` are you referring to? In a test do you mean?

**Comment by juliev0** [comment]
> Fix the Json parser to convert number to int64 instead of float64

In some cases it should be float and in some cases it should be int. It really can't know unfortunately. It makes sense that the json unmarshal just assumes float in that case.

**Comment by juliev0** [comment]
> [Here's](https://github.com/numaproj/numaplane/pull/459) the fix I have raised, Unit test cases is passing will check the E2E test in pipeline, hopefully it should also pass.

I don't think we should do it that way unless I'm mistaken in some way. I added a comment for why. 

Let me think about this.

**Comment by chandankumar4** [comment]
okay, As it's an open question about JsonParser that it treat all numeric value to float64, and maybe people have followed above approach to fix it, but it depend on use-cases. OR we can think about [2nd approach](https://github.com/numaproj/numaplane/issues/456#issuecomment-2545055746) also.
